rialto_airflow/funders/linker.py

```diff
@@ -116,35 +114,81 @@ def _find_or_create_funder(snapshot, funder: dict) -> Optional[int]:
 
     federal = is_federal_grid_id(grid_id) or is_federal(name)
 
-    with get_session(snapshot.database_name).begin() as session:
-        funder_id = session.execute(
-            insert(Funder)
-            .values(name=name, grid_id=grid_id, federal=federal)
-            .on_conflict_do_update(
-                constraint="funder_grid_id_key", set_=dict(name=name, federal=federal)
-            )
-            .returning(Funder.id)
-        ).scalar_one()
+    funder_id = session.execute(
+        insert(Funder)
+        .values(name=name, grid_id=grid_id, federal=federal)
+        .on_conflict_do_update(
+            constraint="funder_grid_id_key", set_=dict(name=name, federal=federal)
+        )
+        .returning(Funder.id)
+    ).scalar_one()
 
     return funder_id
 
 
-def _lookup_openalex_funder(funder: dict) -> Optional[dict]:
+def _find_or_create_openalex_funder(
+    session: Session, openalex_id: str
+) -> Optional[int]:
+    # if the funder is in the database aleady return it
+    funder = session.execute(
+        select(Funder).where(Funder.openalex_id == openalex_id)
+    ).scalar_one_or_none()
+    if funder:
+        return funder.id
```

Comment:
Would be good to add test coverage for this?

Reply:
Good idea. I added one.